File: auth-server/src/main/java/no/vicx/authserver/AuthServerApplication.java
package no.vicx.authserver;

import no.vicx.authserver.config.DefaultUserProperties;
import no.vicx.authserver.config.OAuthProperties;
import no.vicx.database.user.VicxUser;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.autoconfigure.domain.EntityScan;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.data.jpa.repository.config.EnableJpaRepositories;

@SpringBootApplication
@EnableConfigurationProperties({OAuthProperties.class, DefaultUserProperties.class})
@EnableJpaRepositories(basePackageClasses = VicxUser.class)
@EntityScan(basePackageClasses = VicxUser.class)
public class AuthServerApplication {
    public static void main(String[] args) {
        SpringApplication.run(AuthServerApplication.class, args);
    }
}

File: auth-server/src/main/kotlin/no/vicx/authserver/AuthServerApplication.kt
package no.vicx.authserver

import no.vicx.authserver.config.DefaultUserProperties
import no.vicx.authserver.config.OAuthProperties
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.properties.EnableConfigurationProperties

inline fun <reified T : Any> loggerFor(): Logger = LoggerFactory.getLogger(T::class.java)

@SpringBootApplication
@EnableConfigurationProperties(OAuthProperties::class, DefaultUserProperties::class)
class AuthServerApplication

fun main(args: Array<String>) {
    SpringApplication.run(AuthServerApplication::class.java, *args)
}

